models/Ion_auth_model.php

```diff
@@ -1878,23 +1894,35 @@ public function remember_user($identity)
 			return FALSE;
 		}
 
-		// The selector is a simple token to retrieve the user
-		$selector = $this->_random_token(40);
-		// The validator will strictly validate the user and should be more complex
-		$validator = $this->_random_token(128);
-		// Of course, we store the validator hash to avoid session stealing if DB is leaked
-		$validator_hashed = $this->hash_password($validator, $identity);
+		// Generate random tokens
+		$token = $this->_generate_selector_validator_couple();
 
-		if ($validator_hashed)
+		if ($token->validator_hashed)
 		{
 			$this->db->update($this->tables['users'],
-				array('remember_selector' => $selector,
-					  'remember_code' => $validator_hashed),
-				array($this->identity_column => $identity));
+								array('remember_selector' => $token->selector,
+									  'remember_code' => $token->validator_hashed),
+								array($this->identity_column => $identity));
 
 			if ($this->db->affected_rows() > -1)
 			{
-				$this->_set_remember_cookie_info($selector, $validator);
+				// if the user_expire is set to zero we'll set the expiration two years from now.
+				if($this->config->item('user_expire', 'ion_auth') === 0)
+				{
+					$expire = (60*60*24*365*2);
```

Comment:
Change this to 63072000 and add a comment on what it is so we don't always do this in the code for no real reason other than readability.  Yea I know this was my code from before...

Reply:
Agreed